src/NServiceBus.RabbitMQ/Receiving/MessageConverter.cs

```diff
@@ -98,50 +97,65 @@ string DefaultMessageIdStrategy(BasicDeliverEventArgs message)
 
         static string ValueToString(object value)
         {
-            var s = value as string;
-            if (s != null)
-            {
-                return s;
-            }
-
             var bytes = value as byte[];
             if (bytes != null)
             {
                 return Encoding.UTF8.GetString(bytes);
             }
 
-            var dictionary = value as IDictionary<string, object>;
+            var dictionary = value as Dictionary<string, object>;
```

Comment:
Because in that case the underlying type is actually a `Dictionary`. Casting to that instead means we don't end up with a heap allocation for enumerating it via the foreach.